src/main/java/core/basesyntax/service/strategy/ActivityStrategy.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.service.strategy.impl.BalanceService;
+import core.basesyntax.service.strategy.impl.PurchaseService;
+import core.basesyntax.service.strategy.impl.ReturnService;
+import core.basesyntax.service.strategy.impl.SupplyService;
+import core.basesyntax.utility.FruitTransaction;
+
+public class ActivityStrategy {
+    public ActivityService getActivityService(FruitTransaction.Operation activityType) {
+        return switch (activityType) {
```

Comment:
@MolchanovAlexander You should not approve the PR untill a big mistake like that is fixed

Reply:
Agree, we should have a `Map<Operation, OperationHandler>`, initialize it in the main, and pass here through the constructor to get the right handler by operation instead of switch, check the theory and Bohdan's example on why it is better